dingo/gw/noise/synthetic_noise/generate_dataset.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import copy
+import textwrap
+import numpy as np
+from typing import Dict
+
+import yaml
+
+from dingo.gw.dataset import DingoDataset
+from dingo.gw.noise.asd_dataset import ASDDataset
+from dingo.gw.noise.synthetic_noise.asd_parameterization import parameterize_asd_dataset
+from dingo.gw.noise.synthetic_noise.asd_sampling import KDE, get_rescaling_params
+from dingo.gw.noise.synthetic_noise.utils import reconstruct_psds_from_parameters
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=textwrap.dedent(
+            """\
+        Generate a synthetic noise ASD dataset from an existing dataset of real ASDs.
```

Comment:
This help should be a bit more complete: The synthetic ASD dataset also depends on an additional ASD for scaling, and it's not clear that this is the case, or how this should be specified.